tests/test_potentials.py

```diff
@@ -37,6 +37,20 @@ def test_bound_potential_get_potential(harmonic_bond):
     assert unbound_impl is bound_impl.get_potential()
 
 
+def test_bound_potential_set_params(harmonic_bond):
```

Comment:
Test expansion request: To increase peace of mind for future changes, can we increase coverage here?

(Expansion along 2 axes: (1) Cover more Potential classes and signatures (esp. corner cases in Nonbonded such as changing exception parameters, and SummedPotential such as composition with `.call_with_params_list`). (2) Test on more random instances -- may be faster to assert self-consistency with unbound potential (to avoid calling `.to_gpu(np.float32)` for every test instance).)

Reply:
> esp. corner cases in Nonbonded such as changing exception parameters

Could you clarify what you mean here? (assuming you mean exclusion parameters, these are attributes of the underlying `Potential` objects and are not settable using `set_params`)

Reply:
>Could you clarify what you mean here? (assuming you mean exclusion parameters, these are attributes of the underlying Potential objects and are not settable using set_params)

Ahh, sorry -- yes, meant exclusion parameters, but got confused about whether these would be settable or not. This comment is not actionable, since exclusion_scales wouldn't be settable using set_params.

Reply:
OK, thanks - I think the other suggestions for strengthening the tests in the original comment are actionable, though.

Reply:
Thinking about this more, testing with different potentials (even `SummedPotential`) feels redundant, as `set_params` (and `BoundPotential` in general) is agnostic to / polymorphic in `potential`. Will think about it a bit more in case I'm missing something.

Reply:
>Thinking about this more, testing with different potentials (even SummedPotential) feels redundant, as set_params (and BoundPotential in general) is agnostic to / polymorphic in potential. Will think about it a bit more in case I'm missing something.

I don't think it would expose any errors in the current PR, but increased coverage would just increase peace of mind for future changes.

(For example, I could imagine possible complications in how this function interacts with foreseeable optimizations in the nonbonded implementation, and these complications wouldn't necessarily show up in testing the harmonic bond implementation. If (in a future optimization) the neighborlist were computed in a way that was sensitive to both `coords` and `w_coords`, then using `set_params` to update `w_coords` should invalidate the neighborlist. If the test suite exercised only updating a harmonic bond potential's parameters but didn't exercise updating a nonbonded potential's parameters, we might miss if, say, https://github.com/proteneer/timemachine/blob/aecade69b90e6c13aaa82855a50e6210abd724db/timemachine/cpp/src/nonbonded_all_pairs.cu#L201-L208 didn't handle this corner case correctly.)

Reply:
Would it be OK open a separate issue with the above example for follow up?

It seems like any issue that could be exposed by calling `set_params` on a `BoundPotential` would also be observable when calling the unbound potential successively with different parameters. I think adding tests along these lines would be useful, but I would tend to prefer them to be independent of `BoundPotential`.

Reply:
Migrated to https://github.com/proteneer/timemachine/issues/1120 !

(Not intended to be blocking for this PR -- looks good to me as-is.)